ql/src/java/org/apache/hadoop/hive/ql/HiveQueryLifeTimeHook.java

```diff
@@ -71,20 +71,23 @@ private void checkAndRollbackCTAS(QueryLifeTimeHookContext ctx) {
     QueryPlan queryPlan = ctx.getHookContext().getQueryPlan();
     boolean isCTAS = Optional.ofNullable(queryPlan.getQueryProperties())
         .map(queryProps -> queryProps.isCTAS()).orElse(false);
+    // return early if the query is not CATS type.
+    if (!isCTAS)
+      return;
 
     PrivateHookContext pCtx = (PrivateHookContext) ctx.getHookContext();
     Path tblPath = pCtx.getContext().getLocation();
 
-    try {
-      FileSystem fs = tblPath.getFileSystem(conf);
-      if (!fs.exists(tblPath)) {
-        return;
+    if (tblPath != null) {
```

Comment:
I feel its better to include entire logic below - 
```
if (isCTAS && tblPath != null) {
     try {
        FileSystem fs = tblPath.getFileSystem(conf);
        if (!fs.exists(tblPath)) {
          return;
        }
      } catch (Exception e) {
        throw new RuntimeException("Not able to check whether the CTAS table directory exists due to: ", e);
      }
      // Remaining logic
      ....
 }
```
The CTAS check and the non-nullability of the path will be checked then in a single if clause.

Reply:
Changed. Thanks.